code-studio/src/js/progress.js

```diff
@@ -157,14 +158,33 @@ function loadProgress(scriptData, currentLevelId, saveAnswersBeforeNavigation =
         newProgress[key] = clientState.mergeActivityResult(state.progress[key], action.progress[key]);
       });
 
-      return Object.assign({}, state, {
+      const stages = state.stages.map(stage => Object.assign({}, stage, {levels: stage.levels.map(level => {
+        const id = level.uid || progress.bestResultLevelId(level.ids, newProgress);
+
+        return Object.assign({}, level, {
+          status: progress.activityCssClass(newProgress[id]),
+          id: id,
+          url: level.url
+        });
+      })}));
+
+      const returnState = Object.assign({}, state, {
         progress: newProgress,
-        stages: state.stages.map(stage => Object.assign({}, stage, {levels: stage.levels.map(level => {
-          let id = level.uid || progress.bestResultLevelId(level.ids, newProgress);
+        stages: stages
+      });
 
-          return Object.assign({}, level, {status: progress.activityCssClass(newProgress[id])});
-        })}))
+      const peerReviewStage = _.findLast(returnState.stages, function (stage) {
+        return stage.flex_category === 'Peer Review';
       });
+
+      if (action.peerReviewsPerformed) {
+        action.peerReviewsPerformed.forEach(function (peerReview, index) {
+            Object.assign(peerReviewStage.levels[index], peerReview);
```

Comment:
I broke it out this way because stages are composed of levels, but the peer review section is composed of reviews. The above map statement was already hard to parse (ending with })})); ) and I didn't see a good place to insert something that wasn't mapping levels within a stage.